container/container_unix.go

```diff
@@ -302,6 +302,9 @@ func (container *Container) UpdateContainer(hostConfig *containertypes.HostConfi
 
 	// update HostConfig of container
 	if hostConfig.RestartPolicy.Name != "" {
+		if container.HostConfig.AutoRemove {
+			return fmt.Errorf("Restart policy of container with AutoRemove flag is not allowed to update")
```

Comment:
I think error from daemon is enough, no need to translate. 
I like the first sentense. :smile: